ibis/backends/sql/compilers/snowflake.py

```diff
@@ -452,25 +452,36 @@ def visit_TimestampFromUNIX(self, op, *, arg, unit):
         timestamp_units_to_scale = {"s": 0, "ms": 3, "us": 6, "ns": 9}
         return self.f.to_timestamp(arg, timestamp_units_to_scale[unit.short])
 
-    def _array_collect(self, *, arg, where, order_by, include_null):
+    def _array_collect(self, *, arg, where, order_by, include_null, distinct=False):
         if include_null:
             raise com.UnsupportedOperationError(
                 "`include_null=True` is not supported by the snowflake backend"
             )
+        if where is not None and distinct:
+            raise com.UnsupportedOperationError(
+                "Combining `distinct=True` and `where` is not supported by snowflake"
```

Comment:
```suggestion
                "Combining `distinct=True` and `where` is not supported by Snowflake"
```

Only because it's a proper noun and not a generic one :)

Alternatively, we can leave this for a follow up and audit all the backends for proper noun capitalization.

Reply:
I know there's more of these in the codebase, I'll leave this as a follow up.